vm/state_migration/src/lib.rs

```diff
@@ -1,9 +1,51 @@
 // Copyright 2019-2023 ChainSafe Systems
 // SPDX-License-Identifier: Apache-2.0, MIT
 
-pub use nv18::run_migration as run_nv18_migration;
-pub use nv19::run_migration as run_nv19_migration;
+use std::sync::Arc;
+
+use cid::Cid;
+use forest_networks::{ChainConfig, Height};
+use forest_shim::clock::ChainEpoch;
+use forest_utils::miscs::reveal_five_trees;
+use fvm_ipld_blockstore::Blockstore;
 
 pub(crate) mod common;
 mod nv18;
 mod nv19;
+
+type RunMigration<DB> = fn(&ChainConfig, &DB, &Cid, ChainEpoch) -> anyhow::Result<Cid>;
+
+/// Run state migrations
+pub fn run_state_migrations<DB>(
+    epoch: ChainEpoch,
+    chain_config: &Arc<ChainConfig>,
+    db: &DB,
+    parent_state: &Cid,
+) -> anyhow::Result<Option<Cid>>
+where
+    DB: 'static + Blockstore + Clone + Send + Sync,
+{
+    let mappings: [(_, RunMigration<DB>); 2] = [
+        (Height::Hygge, nv18::run_migration::<DB>),
+        (Height::Lightning, nv19::run_migration::<DB>),
+    ];
+
+    for (height, migrate) in mappings {
```

Comment:
Before we had a `match` and now we have a `for` loop. Could this be avoided (i.e. using const code)?

Reply:
I don't think so because `DB` is a generic type, it would require some fancy compile-time reflection to achieve the goal, or any suggestions?